starter-code/pizza.js

```diff
@@ -17,69 +17,154 @@ var state = {
-  renderButtons()
-  renderPrice()
+  renderPepperonni();
+  renderMushrooms();
+  renderGreenPeppers();
+  renderWhiteSauce();
+  renderGlutenFreeCrust();
+
+  renderButtons();
+  renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll(".pep").forEach(function($pep) {
     if (state.pepperonni) {
       $pep.style.visibility = "visible";
-    }
-    else {
+    } else {
       $pep.style.visibility = "hidden";
     }
-  })
+  });
 }
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll(".mushroom").forEach(function($mushroom) {
+    if (state.mushrooms) {
+      $mushroom.style.visibility = "visible";
+    } else {
+      $mushroom.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll(".green-pepper").forEach(function($greenPeppers) {
+    if (state.greenPeppers) {
+      $greenPeppers.style.visibility = "visible";
+    } else {
+      $greenPeppers.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  const whiteSauce = document.querySelector(".sauce");
+  if (state.whiteSauce) {
+    whiteSauce.classList.add("sauce-white");
+  } else {
+    whiteSauce.classList.remove("sauce-white");
+  }
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  const glutenFree = document.querySelector(".crust");
+  if (state.glutenFreeCrust) {
+    glutenFree.classList.add("crust-gluten-free");
+  } else {
+    glutenFree.classList.remove("crust-gluten-free");
+  }
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  if (state.pepperonni) {
+    document.querySelector("button.btn-pepperonni").classList.add("active");
+  } else {
+    document.querySelector("button.btn-pepperonni").classList.remove("active");
+  }
+
+  if (state.mushrooms) {
+    document.querySelector("button.btn-mushrooms").classList.add("active");
+  } else {
+    document.querySelector("button.btn-mushrooms").classList.remove("active");
+  }
+
+  if (state.greenPeppers) {
+    document.querySelector("button.btn-green-peppers").classList.add("active");
+  } else {
+    document
+      .querySelector("button.btn-green-peppers")
+      .classList.remove("active");
+  }
+
+  if (state.whiteSauce) {
+    document.querySelector("button.btn-sauce").classList.add("active");
+  } else {
+    document.querySelector("button.btn-sauce").classList.remove("active");
+  }
+
+  if (state.glutenFreeCrust) {
+    document.querySelector("button.btn-crust").classList.add("active");
+  } else {
+    document.querySelector("button.btn-crust").classList.remove("active");
+  }
 }
```

Comment:
For the render buttons you should have tried the toggle class
something like this:
``buttonElement.classList.toggle("active);``